java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/persistence/DynamoDbMetastoreImpl.java

```diff
@@ -226,39 +233,45 @@ public DynamoDbMetastoreImpl build() {
 
   public interface EndPointStep {
     /**
-     * Adds EndPoint config to the AWS DynamoDb client
-     * @param endPoint the service endpoint either with or without the protocol
-     * @param signingRegion the region to use for SigV4 signing of requests (e.g. us-west-1)
-     * @return The current {@code BuildStep} instance
+     * Adds EndPoint config to the AWS DynamoDb client.
+     *
+     * @param endPoint The service endpoint either with or without the protocol.
+     * @param signingRegion The region to use for SigV4 signing of requests (e.g. us-west-1).
+     * @return The current {@code BuildStep} instance with end point configuration.
      */
     BuildStep withEndPointConfiguration(String endPoint, String signingRegion);
   }
 
   public interface RegionStep {
     /**
-     * Specifies the region for the AWS DynamoDb client
-     * @param region The region for the DynamoDb client
-     * @return The current {@code BuildStep} instance
+     * Specifies the region for the AWS DynamoDb client.
+     *
+     * @param region The region for the DynamoDb client.
+     * @return The current {@code BuildStep} instance with a client region. This overrides the region specified as the
+     *         {@code newBuilder} input parameter.
      */
     BuildStep withRegion(String region);
   }
 
   public interface BuildStep {
     /**
-     * Specifies the name of the table
-     * @param tableName The name of the table
+     * Specifies the name of the table.
+     *
+     * @param tableName A custom name for the table.
      * @return The current {@code BuildStep} instance.
      */
     BuildStep withTableName(String tableName);
 
     /**
-     * Specifies whether key suffix should be enabled for DynamoDB
+     * Specifies whether key suffix should be enabled for DynamoDB. This needs to be enabled for Global Table support.
```

Comment:
Does it make more sense to use "should be enabled" instead of "needs"? Could also expand on the _why_.